SPEC.md

```diff
@@ -2601,6 +2602,20 @@ Integer ylen = length(ys) # 3
 Integer zlen = length(zs) # 0
 ```
 
+## Array[X] flatten(Array[Array[X]])
+
+Given an array of arrays, the `flatten` function concatentes all the
+member arrays in the order to appearance to give the result. For example:
+
+```
+Array[Array[Integer]] ai2D = [[1, 2, 3], [11], [21,22]]
+Array[Integer] ai = flatten(ai2D)   # [1, 2, 3, 11, 21, 22]
+
+
+Array[Array[File]] af2D = [["/tmp/X.txt"], ["/tmp/Y.txt", "/tmp/Z.txt"], []]
+Array[Integer] af = flatten(af2D)   # ["/tmp/X.txt", "/tmp/Y.txt", "/tmp/Z.txt"]
```

Comment:
To be honest, I'm not sure how the second example is different from the first?

Reply:
Also should not be `Integer`

Reply:
I wanted to show an example with an empty `Array` the is eliminated, and also show that it works on `File`s.